client/src/js/components/bhLatestInvoice.js

```diff
@@ -27,7 +27,7 @@ function LatestInvoice(Patient, moment) {
 
   /** getting patient document */
   function startup() {
-    if (!vm.debtorUuid) { return; }
+    if (!vm.debtorUuid || !vm.patientInvoice) { return; }
```

Comment:
Isn't `vm.patientInvoice` always undefined at the beginning of the code?  When does it become defined?  It looks like this code will never run past this function...